comp/core/remoteagentregistry/impl/remoteagentregistry.go

```diff
@@ -142,9 +153,14 @@ func (ra *remoteAgentRegistry) RegisterRemoteAgent(registration *remoteagentregi
 	return recommendedRefreshInterval, nil
 }
 
+func (ra *remoteAgentRegistry) RegisterCollector() {
```

Comment:
I forgot the visibility rules for Go methods, but are we able to make this private and still be able to call it from the test?

It strikes me that we shouldn't allow this be be called directly from anything outside the implementation.

Reply:
Updated the test to call `start` instead 👍